Microsoft/WebTemplateStudio
# some naming convention, file structure, and routing

This PR is to addressed all the remains changes regarding to improve Angular code quality based on Angular experts suggestion ( mainly from #668). The main test point for this PR, is the generated project works and generated code follows name convention. 

closes #668

**Summary:** 
1. naming conventions
- nav-bar
- footer
- grid subfolders
- master detail subfolders
- list subfolders
(some folders still camel case or has _, this will be fixed through change in CoreTS, not in this PR)

2. fix file structure (component should 4 files)
3. add routing files and lazy loading 
4. @output properties should not be suffixed with EventEmitter (just renaming)

**How to test:**
1. generate an angular project with 4 type of pages selected 
2. start the project, the project should start properly
3. run "npm run lint" or "yarn lint" should only has 2 errors (related to master detail, which will be fixed by change in CoreTS, not in this PR)

**Expected:**
![index1](https://user-images.githubusercontent.com/12874317/59315256-d850ab00-8c6d-11e9-8424-5dc4c4bcc9c7.PNG)

![index2](https://user-images.githubusercontent.com/12874317/59315259-da1a6e80-8c6d-11e9-8079-f2634073af6d.PNG)
